File: app/src/main/java/com/awesome/demos/MainActivity.java
package com.awesome.demos;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.awesome.designpatterndemo.User;

/**
 * Author: JfangZ
 * Email: zhangjingfang@jeejio.com
 * Date: 2021/4/13 17:44
 * Description:
 */
public class MainActivity extends AppCompatActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        User.observer3();
//        getLifecycle().addObserver(new MyLifeCycleObserver());
    }


    @Override
    protected void onStart() {
        super.onStart();
    }

    @Override
    protected void onPause() {
        super.onPause();

    }

    @Override
    protected void onStop() {
        super.onStop();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
    }
}

File: app/src/main/java/com/awesome/demos/MainActivity.kt
package com.awesome.demos

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.awesome.javadomo.threads.ThreadUtil

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        ThreadUtil.scheduledThreadPool()
        ThreadUtil.singleThreadScheduledExecutor()
        // 子线程跳转界面
//        findViewById<View>(R.id.tv).setOnClickListener { Thread(Runnable { startActivity(Intent(this@MainActivity, TextViewLinkActivity::class.java)) }).start() }
        findViewById<View>(R.id.tv).setOnClickListener { Toast.makeText(this, "sds", Toast.LENGTH_LONG).show()}
    }
}